demo/repo/Jenkinsfile

```diff
@@ -17,26 +17,31 @@ parallel(longerTests: {
     runTests(servers, 20)
 })
 
-stage name: 'Staging', concurrency: 1
-node {
-    servers.deploy 'staging'
-}
+milestone ordinal: 1
+lock (resource: 'staging-server', inversePrecedence: true) {
+    stage name: 'Staging'
```

Comment:
Yeah, I'm changing the previous behavior. IMO the build is not really *in* _Staging_ until it acquires the lock.

> (The stage view will show delays explicitly if you add PauseAction from the lock step.

Thanks for the interesting tip! I'll add this to `lock` (but not now).